pipenv/utils.py

```diff
@@ -234,9 +234,10 @@ def get_resolver_metadata(deps, index_lookup, markers_lookup, project, sources):
         dep = " ".join(remainder)
         req = Requirement.from_line(dep)
         constraints.append(req.constraint_line)
-
         if url:
-            index_lookup[req.name] = project.get_source(url=url).get("name")
+            source = first(s for s in sources if url.startswith(s.get("url")))
```

Comment:
Is there a chance that source has no url field? ```url.startswith(None)``` raises exceptions.

Reply:
Updated

Reply:
:+1: